upup/pkg/fi/cloudup/defaults.go

```diff
@@ -109,6 +94,27 @@ func PerformAssignments(c *kops.Cluster, vfsContext *vfs.VFSContext, cloud fi.Cl
 		}
 	}
 
+	if cloud.ProviderID() == kops.CloudProviderAzure && c.Spec.Networking.NetworkCIDR == "" {
+		if c.SharedVPC() {
+			if c.Spec.CloudProvider.Azure == nil || c.Spec.CloudProvider.Azure.ResourceGroupName == "" {
+				return fmt.Errorf("missing required --azure-resource-group-name when specifying Network ID")
+			}
+			vpcInfo, err := cloud.(azure.AzureCloud).FindVNetInfo(c.Spec.Networking.NetworkID, c.Spec.CloudProvider.Azure.ResourceGroupName)
+			if err != nil {
+				return err
+			}
+			if vpcInfo == nil {
+				return fmt.Errorf("unable to find Network ID %q", c.Spec.Networking.NetworkID)
+			}
+			c.Spec.Networking.NetworkCIDR = vpcInfo.CIDR
+			if c.Spec.Networking.NetworkCIDR == "" {
+				return fmt.Errorf("unable to infer NetworkCIDR from Network ID, please specify --network-cidr")
+			}
+		} else {
+			c.Spec.Networking.NetworkCIDR = "10.0.0.0/16"
```

Comment:
Don't want to use a const for this ?

Reply:
I will do a cleanup there in a separate PR, for now don't want to change the style.